osidb/migrations/0081_migrate_flawacknowledgment_flawmeta.py

```diff
@@ -0,0 +1,54 @@
+# Written manually on 2023-07-07
+
+# https://docs.djangoproject.com/en/4.2/ref/migration-operations/
+from django.db import migrations
+from ast import literal_eval
+
+def forwards_func(apps, schema_editor):
+    FlawMeta = apps.get_model("osidb", "FlawMeta")
+    FlawAcknowledgment = apps.get_model("osidb", "FlawAcknowledgment")
+    for meta_ack in FlawMeta.objects.filter(type="ACKNOWLEDGMENT"):
+        flaw = meta_ack.flaw
+        name = meta_ack.meta_attr["name"]
+        # Convert None to empty string.
+        affiliation = meta_ack.meta_attr.get("affiliation") or ""
+        # hstore holds bolean values as strings containing True|False
+        # so we need to explicitly convert it to the bolean value
+        from_upstream = literal_eval(meta_ack.meta_attr["from_upstream"])
+
+        # FlawAcknowledgment that already exist are skipped because they
+        # are exactly as up-to-date as FlawMeta ACKNOWLEDGMENT.
+
+        if not FlawAcknowledgment.objects.filter( flaw=flaw, name=name, affiliation=affiliation ).exists():
+            FlawAcknowledgment.objects.create(
+                flaw=flaw,
+                name=name,
+                affiliation=affiliation,
+                from_upstream=from_upstream,
+                meta_attr=meta_ack.meta_attr,
+                acl_read=meta_ack.acl_read,
+                acl_write=meta_ack.acl_write,
+                created_dt=meta_ack.created_dt,
+                updated_dt=meta_ack.updated_dt
+            )
+
+
+def reverse_func(apps, schema_editor):
+    # forwards_func() creates FlawAcknowledgment instances that mirror
+    # FlawMeta ACKNOWLEDGMENT instances that are identical to FlawAcknowledgment
+    # that would be created using Flaw bzimport, and that could have already
+    # been created during Flaw bzimport before forwards_func() executed.
+    # No way to revert, and no reason to revert.
+    # You might consider reverting back to 0078 which is before FlawAcknowledgment
+    # was created.
+    pass
+
+
+class Migration(migrations.Migration):
+    dependencies = [
+        ('osidb', '0080_alter_flawacknowledgment_affiliation'),
+    ]
+
+    operations = [
+        migrations.RunPython(forwards_func, reverse_func),
```

Comment:
I believe this operation should be a transaction, I think `atomic=True` would do the trick, what do you think?

Reply:
Didn't know about that! Thanks! That would be bad, esp. in the hypothetical case where osidb runs past the partially failed migration and doesn't retry it / marks it as ok. I don't know enough about that, but I added `atomic=True` and locally tweaked the code to make the migration fail for some model instances and `make migrate` fails loudly and django never marks the migration as successful, so now it should be fine.

Tested on production data dump from February 2023.

Reply:
(Fixed in a rebased commit. IMHO resolved.)